localstack-core/localstack/services/lambda_/provider.py

```diff
@@ -1805,23 +1805,41 @@ def create_event_source_mapping_v1(
         # TODO: find a better abstraction to create these
         params = request.copy()
         params.pop("FunctionName")
-        batch_size = api_utils.validate_and_set_batch_size(
-            request["EventSourceArn"], request.get("BatchSize")
-        )
+        if not (service_type := self.get_source_type_from_request(request)):
+            raise InvalidParameterValueException("Unrecognized event source.")
+
+        batch_size = api_utils.validate_and_set_batch_size(service_type, request.get("BatchSize"))
         params["FunctionArn"] = fn_arn
         params["BatchSize"] = batch_size
         params["UUID"] = new_uuid
         params["MaximumBatchingWindowInSeconds"] = request.get("MaximumBatchingWindowInSeconds", 0)
         params["LastModified"] = api_utils.generate_lambda_date()
         params["FunctionResponseTypes"] = request.get("FunctionResponseTypes", [])
         params["State"] = "Enabled"
-        if "sqs" in request["EventSourceArn"]:
+        if "sqs" in service_type:
             params["StateTransitionReason"] = "USER_INITIATED"
             if batch_size > 10 and request.get("MaximumBatchingWindowInSeconds", 0) == 0:
                 raise InvalidParameterValueException(
                     "Maximum batch window in seconds must be greater than 0 if maximum batch size is greater than 10",
                     Type="User",
                 )
+        elif service_type == "kafka":
+            params["StartingPosition"] = request.get("StartingPosition", "TRIM_HORIZON")
+            params["StateTransitionReason"] = "USER_INITIATED"
```

Comment:
Such state updates are unrelated to kafka and should ultimately handled differently. I'm okay with hardcoding these for ESM v1, but we should revisit this in ESM v2.
/cc @gregfurman